backend/clap/src/main/java/zpi/ppea/clap/repository/PartialRatingRepository.java

```diff
@@ -0,0 +1,31 @@
+package zpi.ppea.clap.repository;
+
+import data_store.DataStoreGrpc;
+import data_store.PartialRating;
+import lombok.RequiredArgsConstructor;
+import net.devh.boot.grpc.client.inject.GrpcClient;
+import org.springframework.stereotype.Repository;
+import zpi.ppea.clap.dtos.UpdatePartialRatingDto;
+import zpi.ppea.clap.exceptions.GrpcDatastoreException;
+import zpi.ppea.clap.mappers.PartialRatingMapper;
+import zpi.ppea.clap.security.FirebaseAgent;
+
+import java.util.concurrent.ExecutionException;
+
+@Repository
+@RequiredArgsConstructor
+public class PartialRatingRepository {
+
+    @GrpcClient("dataStore")
+    DataStoreGrpc.DataStoreFutureStub dataStoreFutureStub;
+
+    public PartialRating upsertPartialRating(UpdatePartialRatingDto updatePartialRatingDto, FirebaseAgent.UserAuthData authentication) {
+        try {
+            return dataStoreFutureStub.postPartialRating(PartialRatingMapper.dtoToPartialRatingRequest(
+                    updatePartialRatingDto, authentication.getClaims().getAssessorId())).get();
+        } catch (InterruptedException | ExecutionException e) {
+            Thread.currentThread().interrupt();
```

Comment:
What is the point of interrupting current thread... after it caught interrupt exception?

Reply:
Rule: _java:S2142_ Sonarqube plugin shows this as warning. I found some article about it: https://medium.com/@thaipd/interruptexception-in-java-85483e3c7243

Reply:
We shouldn't leave _InterruptedException_ in catch without action, we should rethrow it or interrupt a thread.

Reply:
Alright, didn't know that. Seems reasonable